src/osd/OSD.cc

```diff
@@ -8647,6 +8647,36 @@ int OSD::init_op_flags(OpRequestRef& op)
     if (ceph_osd_op_mode_cache(iter->op.op))
       op->set_cache();
 
+    // check for ec base pool
+    int64_t poolid = m->get_pg().pool();
+    const pg_pool_t *pool = osdmap->get_pg_pool(poolid);
+    if (pool->is_tier()) {
+      const pg_pool_t *base_pool = osdmap->get_pg_pool(pool->tier_of);
+      assert(base_pool);
+      if (iter->op.op == CEPH_OSD_OP_SYNC_READ ||
+          iter->op.op == CEPH_OSD_OP_MAPEXT ||
+          iter->op.op == CEPH_OSD_OP_SPARSE_READ ||
+          iter->op.op == CEPH_OSD_OP_WRITE ||
+          iter->op.op == CEPH_OSD_OP_ZERO ||
+          iter->op.op == CEPH_OSD_OP_TRUNCATE ||
+          iter->op.op == CEPH_OSD_OP_CLONERANGE ||
+          iter->op.op == CEPH_OSD_OP_TMAPGET ||
+          iter->op.op == CEPH_OSD_OP_TMAPPUT ||
+          iter->op.op == CEPH_OSD_OP_TMAPUP) {
+        if (base_pool->require_rollback()) {
```

Comment:
Eh, actually a helper is probably overkill since this method is the only user, *and* it may need to take additional information into account (like the whole OSDOp).

I still think flipping this around to be a whitelist is an improvement, though!